doc/cephadm/services/osd.rst

```diff
@@ -245,6 +245,18 @@ Expected output::
 
 OSDs that are not safe to destroy will be rejected.
 
+.. note::
+    After removing OSDs, if the drives the OSDs were deployed on once again
+    become available, cephadm may automatically try to deploy more OSDs
+    on these drives if they match an existing drivegroup spec. If you deployed
+    the OSDs you are removing with a spec and don't want any new OSDs
+    deployed on the drives after removal, it's best to modify the drivegroup spec.
```

Comment:
Perhaps be clear about ordering with

"modify the drivegroup spec before removal"

Is the default behavior such that if one removes an unfailed OSD without pulling the drive, cephadm will automagically and immediately redeploy it?  If so IMHO the  unmanaged value should be the default for the principle of least astonishment.

Reply:
yeah, the default behavior is to just regularly check for drives that match the spec and pick them up for OSDs. We can't just set unmanaged to true by default because then no OSDs would be created at all and the spec would be useless until a user manually made it no longer unmanaged. What we really need is either for users to set it to unmanaged when they no longer want any more OSDs created, reate some command that provides an easy way to quickly set unmanaged to true/false without needing to manually edit it, or some sort of way to make the drive group spec only attempt to pick up OSDs a single time then mark itself unmanaged. But we'd have to implement it still, so for now what's written here is accurate to the behavior (and has been since cephadm first released back in Octopus afaik).